themes/bootstrap3/templates/RecordDriver/EDS/core.phtml

```diff
@@ -23,36 +23,37 @@
           </a>
         </span>
       <?php endif; ?>
-      <?php $ftLink = $this->driver->getLinkedFullTextLink();
-          if ($ftLink): ?>
-        <span>
-          <a href="<?=$ftLink?>" class="fulltext">
-            <?=$this->transEsc('Linked Full Text')?>
-          </a>
-        </span>
-      <?php endif; ?>
-      <?php $pdfLink = $this->driver->getPdfLink();
-          if ($pdfLink): ?>
-        <span>
-          <a href="<?=$pdfLink?>" class="icon--eds pdf fulltext">
-            <?=$this->transEsc('PDF Full Text')?>
-          </a>
-        </span>
-      <?php endif; ?>
-      <?php $epubLink = $this->driver->getEpubLink();
-          if ($epubLink): ?>
-        <span>
-          <a href="<?=$epubLink?>" class="icon--eds epub fulltext">
-            <?=$this->transEsc('ePub Full Text')?>
-          </a>
-        </span>
-      <?php endif; ?>
-      <?php if ($this->driver->hasHTMLFullTextAvailable()): ?>
-        <span>
-          <a href="<?=$this->escapeHtmlAttr($this->recordLinker()->getUrl($this->driver))?>#html" class="icon--eds html fulltext">
-            <?=$this->transEsc('HTML Full Text')?>
-          </a>
-        </span>
+      <?php if ($this->permission()->isAuthorized('access.EDSExtendedResults')): ?>
+        <?php if ($this->driver->hasHTMLFullTextAvailable()): ?>
+          <span>
+            <a href="<?=$this->escapeHtmlAttr($this->recordLinker()->getUrl($this->driver))?>#html" class="icon--eds html fulltext" target="_blank">
+              <?=$this->transEsc('HTML Full Text')?>
+            </a>
+          </span>
+        <?php endif; ?>
+        <?php if ($ftLink = $this->driver->getLinkedFullTextLink()): ?>
+          <span>
+            <a href="<?=$ftLink?>" class="fulltext" target="_blank">
+              <?=$this->transEsc('Linked Full Text')?>
+            </a>
+          </span>
+        <?php endif; ?>
+        <?php if ($pdfLink = $this->driver->getPdfLink()): ?>
+          <span>
+            <a href="<?=$pdfLink?>" class="icon--eds pdf fulltext" target="_blank">
+              <?=$this->transEsc('PDF Full Text')?>
+            </a>
+          </span>
+        <?php endif; ?>
+        <?php if ($epubLink = $this->driver->getEpubLink()): ?>
+          <span>
+            <a href="<?=$epubLink?>" class="icon--eds epub fulltext" target="_blank">
+              <?=$this->transEsc('ePub Full Text')?>
+            </a>
+          </span>
+        <?php endif; ?>
+      <?php else: ?>
+        <?=$this->render('RecordDriver/EDS/result-list/full-text-links.phtml', ['driver' => $this->driver, 'recordLinker' => $this->recordLinker()])?>
```

Comment:
If we are able to use the result list's full-text-links here on the record page, is it possible to remove the entire `if` block and always do that?  I'm just not clear on when we have to use $this->driver->getXYZLink() rather than $recordLinker->getTabUrl.

Reply:
I tried that and it didn't work. I think it may indicate that the result-list API call returns less information than the record view API call, but I'm short on time and didn't dig deeper into it... so I went for what seemed like a quicker win.

Basically the idea is that the recordLinker method should always work, based on sparse metadata that's always available. The getXYZLink method only works when rich metadata is present, which is dependent on authorization (and also possibly on context).

Reply:
So purely coincidentally, I've been looking into a bug reported this morning on our EDS record pages, where we have a legacy customization that generates a PDF link using the recordLinker, and it does *not* work.  I haven't figured out why and it's possibly unrelated, but I need to dig into that before agreeing that it's a safer approach.

Reply:
Should I review this branch from the public-facing end, or should I wait till you two get further along on the code?

Reply:
> I haven't figured out why and it's possibly unrelated, but I need to dig into that before agreeing that it's a safer approach.

Ok never mind my concern there, it's just a bug in our legacy code (or more specifically something not kept up to date with changes based on how it was originally done).  I will do the code review.  Defer to you and @sturkel89 of course on when to do the FE testing.

Reply:
Sorry of this is obvious, but I'm not sure what you mean by "FE testing." :-)

And stepping back a bit, I should clarify my earlier comments a little bit.

You asked why we don't just use the recordlinker approach everywhere, and I answered that it didn't work. That's not entirely accurate. What I tried to do was move the entire if/else logic from core.phtml into full-text-links.phtml, so we could use the smarter logged-in links in the search results when appropriate. THAT'S the thing that didn't work. When logged in, I got no links at all, so I had to roll back to using the recordlinker approach in search results all the time.

So then the question comes down to whether or not it's beneficial to use the "smarter" links on the record page when logged in. In theory, this gets the user to the content more directly since it skips the need to jump through a VuFind-mediated login process. It also eliminates links for things that don't really exist. But right now, those non-existent links still work -- I believe we have code that, if you ask for one type of eBook and it doesn't exist, will just forward you along to the next best thing it can find. That's why those phantom PDF links in Susan's example ultimately just send you to the ePub link.

Reply:
> I tried that and it didn't work. I think it may indicate that the result-list API call returns less information than the record view API call, but I'm short on time and didn't dig deeper into it... 

Ok yes you are correct about this.  The result-list API does not return the actual link (to be available to `getPdfLink()` even when there *is* a PDF and you are logged-in and authorized for it.  It must be some performance optimization on the EDS side to not gather all those links, although I find it hard to believe that the link is not as easily available as the rest of the metadata.

Reply:
Thanks for confirming that! I think our options are pretty clear, but which one to pick remains open to debate. I think the work I have done here is probably a step toward a better experience, but it's an open question whether to take it further.

Reply:
> So then the question comes down to whether or not it's beneficial to use the "smarter" links on the record page when logged in. In theory, this gets the user to the content more directly since it skips the need to jump through a VuFind-mediated login process. It also eliminates links for things that don't really exist. But right now, those non-existent links still work -- I believe we have code that, if you ask for one type of eBook and it doesn't exist, will just forward you along to the next best thing it can find. That's why those phantom PDF links in Susan's example ultimately just send you to the ePub link.

> So do we prefer "technically faster/more accurate some of the time" or "consistently wrong, but more-or-less functional all the time"?

> ...  I think our options are pretty clear, but which one to pick remains open to debate. I think the work I have done here is probably a step toward a better experience, but it's an open question whether to take it further.

I am torn on this as well.  Lehigh has a legacy customization page that does use the recordLinker all the time on the EDS record page.  @mis306lu may have had this scenario in mind at the time.  So we are coming from the opposite starting point and it's hard for me to evaluate.  But I agree that the change you made is at least an incremental improvement to the way the VF dev branch handles things right now.